### DISCUSSION TITLE
Zeek on fips

### DISCUSSION THREAD

**Discussion by reyesj2** [zeek/zeek#3377] [Ideas]
Disclaimer: Not a dev in the slightest..

Is something like this possible?

Currently, it looks like the hash algorithms are initialized in [digest.cc](https://github.com/zeek/zeek/blob/release/6.0/src/digest.cc)
Specifically
``` cpp
EVP_MD_CTX* hash_init(HashAlgorithm alg)
	{
	EVP_MD_CTX* c = EVP_MD_CTX_new();
	const EVP_MD* md;

	switch ( alg )
		{
		case Hash_MD5:
#ifdef EVP_MD_CTX_FLAG_NON_FIPS_ALLOW
			/* Allow this to work even if FIPS disables it */
			EVP_MD_CTX_set_flags(c, EVP_MD_CTX_FLAG_NON_FIPS_ALLOW);
#endif
			md = EVP_md5();
			break;
```
Previously the NON_FIPS_ALLOW flag was used to bypass fips restrictions for md5
This no longer works and triggers EVP_DigestInit failed

Can it be updated to use the fips=no argument from https://www.openssl.org/docs/OpenSSL300Design.html under Multilevel Queries?

```cpp
EVP_MD_CTX* hash_init(HashAlgorithm alg)
    {

    EVP_MD_CTX* c = EVP_MD_CTX_new();
    const EVP_MD* md;

    switch ( alg )
        {
        case Hash_MD5:
            md = EVP_MD_fetch(NULL, "MD5", "fips=no");
            break;
```

Trying to test locally, but again I am no dev and would think there is more to do than change just one function. I saw there are references to having to use EVP_MD_free() after, but wouldn't know where to start tracking that down

**Comment by timwoj** [comment]
There was some discussion regarding FIPS in https://github.com/zeek/zeek/issues/2505. It'd be best to move this over to that issue, since we're still trying to figure out what exactly to do about it. One suggestion was to just remove MD5 support, but nothing was ever fully decided.
